neurodamus/node.py

```diff
@@ -1536,30 +1533,29 @@ def cleanup(self):
             self.clear_model(avoid_creating_objs=True)
 
         if SimConfig.delete_corenrn_data and not SimConfig.dry_run:
-            data_folder = SimConfig.coreneuron_datadir
-            logging.info("Deleting intermediate data in %s", data_folder)
-
             with timeit(name="Delete corenrn data"):
-                if MPI.rank == 0:
-                    if ospath.islink(data_folder):
-                        # in restore, coreneuron data is a symbolic link
-                        os.unlink(data_folder)
-                    else:
-                        subprocess.call(['/bin/rm', '-rf', data_folder])
-                    os.remove(ospath.join(SimConfig.output_root, "sim.conf"))
-                    if self._run_conf["EnableReports"]:
-                        os.remove(ospath.join(SimConfig.output_root, "report.conf"))
-
-                # Delete the SHM folder if it was used
-                if self._shm_enabled:
-                    data_folder_shm = SHMUtil.get_datadir_shm(data_folder)
-                    logging.info("Deleting intermediate SHM data in %s", data_folder_shm)
-                    subprocess.call(['/bin/rm', '-rf', data_folder_shm])
+                self._delete_corenrn_data()
+                MPI.barrier()
 
-            MPI.barrier()
+    @run_only_rank0
+    def _delete_corenrn_data(self):
+        data_folder = SimConfig.coreneuron_datadir
+        logging.info("Deleting intermediate data in %s", data_folder)
+
+        if ospath.islink(data_folder):
+            # in restore, coreneuron data is a symbolic link
+            os.unlink(data_folder)
+        else:
+            subprocess.call(['/bin/rm', '-rf', data_folder])
```

Comment:
Is the fork process created here? From the failed job log, it seems we either set `RDMAV_FORK_SAFE` or delete the foler without fork(). For this test purpose, it seems `RDMAV_FORK_SAFE` is sufficient. 
```
For the libfabric EFA provider to work safely when fork()
is called, you will need to set the following environment
variable:
          RDMAV_FORK_SAFE

However, setting this environment variable can result in
signficant performance impact to your application due to
increased cost of memory registration.

You may want to check with your application vendor to see
if an application-level alternative (of not using fork)
exists.
```

Reply:
Why use `subprocess.call` and not something like https://docs.python.org/3/library/shutil.html#shutil.rmtree ?

Reply:
I recalled that we used `shutil.rmtree` at the beginning but it was too slow to delete a folder with a large amount of files https://bbpteam.epfl.ch/project/issues/browse/BBPBGLIB-540 . Then we changed to use "rm -rf" from `subprocess.call`.